libs/angular-components/community/components/overlay/modal/modal.component.ts

```diff
@@ -26,22 +29,34 @@ export type DialogData = ComponentInputs<
   imports: [DialogModule],
   encapsulation: ViewEncapsulation.None,
   host: {
-    "[class.tedi-modal]": "true",
-    "[class.tedi-modal--lg]": "maxWidth() === 'lg'",
-    "[class.tedi-modal--md]": "maxWidth() === 'md'",
-    "[class.tedi-modal--sm]": "maxWidth() === 'sm'",
-    "[class.tedi-modal--small]": "variant() === 'small'",
+    "[class]": "hostClasses()",
   },
 })
 export class ModalComponent implements OnInit {
-  readonly maxWidth = model<ModalSizes>("sm");
+  readonly maxWidth = model<Breakpoint>("sm");
   readonly variant = model<"default" | "small">("default");
 
   readonly dialogRef = inject(DialogRef, { optional: true });
   readonly dialogData = inject(DIALOG_DATA, {
     optional: true,
   });
 
+  // host classes
+  readonly hostClasses = computed(() => {
+    const classes = ["tedi-modal"];
+
+    if (this.variant() === "small") {
+      classes.push("tedi-modal--small");
+    }
+
+    for (const breakpoint of Object.keys(BREAKPOINTS)) {
```

Comment:
i wouldn't take these from breakpoints since these are modal specific sized (for example there's xxl in breapoints, but no such modal max width)